assets/js/hooks/useFormValue.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Form value hook.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { CORE_FORMS } from '../googlesitekit/datastore/forms/constants';
+
+const useFormValue = ( formName, key, defaulValue ) => {
+	return useSelect(
+		( select ) => {
+			const { getValue } = select( CORE_FORMS );
+
+			const value = getValue( formName, key );
+
+			if ( value === undefined && defaulValue !== undefined ) {
+				return defaulValue;
+			}
```

Comment:
`defaultValue` is used in the select function but not in the dependency array – I'm surprised this isn't throwing a lint error.

It doesn't need to be in the select function though. Ideally, the default is handled outside of this hook. Looking at the issue again, this is implied by the AC which calls for only accepting the two parameters which makes more sense, so this should be removed – supporting a default should be an edge case anyways. It's also more clear what is going on to use outside of the hook rather than as a mystery 3rd parameter. e.g. `useFormValue(form, key) ?? defaultValue` should be sufficient for 99% of cases.

There's only a handful of simple instances where the default is used though so this should be a simple change.

Reply:
Thank you @aaemnnosttv for your feedback. I have removed the  `defaultValue` as you suggested and you are right that for most of the instances, it is not required. 

I have reverted couple of instances in `assets/js/components/KeyMetrics/ChipTabGroup/index.js` to keep using `useSelect` because if we use `useFormValue(form, key) ?? defaultValue`, it increases the cyclomatic complexity for the file and we may have to suppress that eslint rule.

Reply:
Thanks! About the `ChipTabGroup`, this component is badly in need of simplification, I'm surprised its within the limit of complexity as it is. Let's use the new form hook even if that requires pushing the limit up temporarily. We need an issue to address this anyways and the new hook is simple enough that we should use it everywhere. We shouldn't ever _disable_ the rule though, it only needs to have its limit updated for the current function as we have before.